File: ahma0/PGS_161989.java
class Solution {

    public int solution(int n, int m, int[] section) {
        int max = 0;
        int cnt = 0;

        for (int i : section) {
            if (max <= i) {
                max = i + m;
                cnt++;
            }
        }

        return cnt;
    }
}

File: julia8024/PGS_161989.py
# 프로그래머스
# Lv1 덧칠하기

def solution(n, m, section):
    answer = 0
    end = 0
    
    for start in section:
        if start > end:
            answer += 1
            end = start + m - 1
        
    return answer
